File: maker/csharp/DbMaker/HeaderBlock.cs
namespace DbMaker
{
    /// <summary>
    ///     header block class
    /// </summary>
    public class HeaderBlock
    {
        public HeaderBlock(long indexStartIp, int indexPtr)
        {
            IndexStartIp = indexStartIp;
            IndexPtr = indexPtr;
        }

        /// <summary>
        ///     index block start ip address
        /// </summary>
        public long IndexStartIp { get; set; }

        /// <summary>
        ///     ip address
        /// </summary>
        public int IndexPtr { get; set; }

        public HeaderBlock SetIndexStartIp(long indexStartIp)
        {
            IndexStartIp = indexStartIp;
            return this;
        }

        public HeaderBlock SetIndexPtr(int indexPtr)
        {
            IndexPtr = indexPtr;
            return this;
        }

        /// <summary>
        ///     get the bytes for db storage
        /// </summary>
        /// <returns></returns>
        public byte[] GetBytes()
        {
            /*
             * +------------+-----------+
             * | 4bytes        | 4bytes    |
             * +------------+-----------+
             *  start ip      index ptr
            */

            var b = new byte[8];

            Util.writeIntLong(b, 0, IndexStartIp);
            Util.writeIntLong(b, 4, IndexPtr);

            return b;
        }
    }
}

File: maker/java/src/org/lionsoul/ip2region/HeaderBlock.java
package org.lionsoul.ip2region;

/**
 * header block class
 * 
 * @author chenxin<chenxin619315@gmail.com>
*/
public class HeaderBlock 
{
    /**
     * index block start ip address
    */
    private long indexStartIp;
    
    /**
     * ip address 
    */
    private int indexPtr;
    
    public HeaderBlock( long indexStartIp, int indexPtr )
    {
        this.indexStartIp = indexStartIp;
        this.indexPtr = indexPtr;
    }

    public long getIndexStartIp()
    {
        return indexStartIp;
    }

    public HeaderBlock setIndexStartIp(long indexStartIp)
    {
        this.indexStartIp = indexStartIp;
        return this;
    }

    public int getIndexPtr()
    {
        return indexPtr;
    }

    public HeaderBlock setIndexPtr(int indexPtr)
    {
        this.indexPtr = indexPtr;
        return this;
    }
    
    /**
     * get the bytes for db storage
     * 
     * @return    byte[]
    */
    public byte[] getBytes()
    {
        /*
         * +------------+-----------+
         * | 4bytes        | 4bytes    |
         * +------------+-----------+
         *  start ip      index ptr
        */
        byte[] b = new byte[8];
        
        Util.writeIntLong(b, 0, indexStartIp);
        Util.writeIntLong(b, 4, indexPtr);
        
        return b;
    }
}
